src/lib/spinel/radio_spinel.hpp

```diff
@@ -1218,9 +1218,9 @@ class RadioSpinel : private Logger
     otMacKey     mPrevKey;
     otMacKey     mCurrKey;
     otMacKey     mNextKey;
-    uint16_t     mSrcMatchShortEntries[OPENTHREAD_CONFIG_MLE_MAX_CHILDREN];
+    uint16_t     mSrcMatchShortEntries[OPENTHREAD_SPINEL_CONFIG_MAX_SRC_MATCH_ENTRIES];
     int16_t      mSrcMatchShortEntryCount;
-    otExtAddress mSrcMatchExtEntries[OPENTHREAD_CONFIG_MLE_MAX_CHILDREN];
+    otExtAddress mSrcMatchExtEntries[OPENTHREAD_SPINEL_CONFIG_MAX_SRC_MATCH_ENTRIES];
```

Comment:
How about adding a `static_assert` here to ensure that if `OPENTHREAD_SPINEL_CONFIG_MAX_SRC_MATCH_ENTRIES` is manually configured, it is indeed larger than `MLE_MAX_CHILDREN` (to detect any misconfigurations):

```c++
    static_assert(OPENTHREAD_SPINEL_CONFIG_MAX_SRC_MATCH_ENTRIES >= OPENTHREAD_CONFIG_MLE_MAX_CHILDREN,
                  "SPINEL_CONFIG_MAX_SRC_MATCH_ENTRIES is not large enough to cover MLE_MAX_CHILDREN");
```